Userland/Libraries/LibSQL/AST/Parser.cpp

```diff
@@ -219,8 +219,10 @@ NonnullRefPtr<Insert> Parser::parse_insert_statement(RefPtr<CommonTableExpressio
     if (consume_if(TokenType::Values)) {
         parse_comma_separated_list(false, [&]() {
             if (auto chained_expression = parse_chained_expression()) {
-                auto chained_expr = dynamic_cast<ChainedExpression*>(chained_expression.ptr());
-                if ((column_names.size() > 0) && (chained_expr->expressions().size() != column_names.size())) {
+                auto* chained_expr = dynamic_cast<ChainedExpression*>(chained_expression.ptr());
+                if (!chained_expr) {
+                    expected("Chained expression"sv);
```

Comment:
I don't think this is the right fix, as `parsed_chain_expression` can clearly return an EXISTS expression. That might have a misunderstanding of the grammar though.

IIRC, the ambiguity was that the grammar here is basically `(expr, expr, expr, ...)` where `expr` is any of these:
https://sqlite.org/syntax/expr.html

So I think at the time, this was trying to disambiguate a single-chain chained expression - `(expr)` - from an EXISTS expression that does not have the `EXISTS` keyword, which is legal - `(SELECT ...)`. However, looking a bit more closely, I think that would require a second set of parentheses, `((SELECT ...))`.

If that's the case, the fix is to probably just remove the ambiguity check from `parsed_chain_expression`.

Reply:
Thanks for the detailed explanation. I've attempted a more complete fix to the problem. 

The fact that you can drop the EXISTS keyword in an ExistsExprerssion means that there is an ambiguity in the grammar at the point that you encounter the second open parenthesis in `((SELECT ...)). In this situation, it is not possible to determine whether you are looking at a nested ChainedExpression or an ExistsExpression before consuming the ParenOpen token. 

The attempt I've made to fix it seems a bit hacky to me. Situations like this could be handled more neatly if the parser had the ability to lookahead multiple tokens. I realize that this isn't straightforward and potentially makes the parser code more difficult to reason about though. Given the few places where ambiguities like this occur, it probably isn't worth doing.